File: viajes365-backend/Viajes365RestApi/Entities/Location.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Viajes365RestApi.Entities
{
    public class Location : Base
    {
        [Key]
        public long LocationId { get; set; }
        [StringLength(100)]
        [Required]
        public string LocationName { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        [StringLength(150)]
        public string FullAddress { get; set; }
        public string Note { get; set; }

        [ForeignKey("FK_Locations_Cities_CityId")]
        public long CityId { get; set; }
        public virtual City City { get; set; }

        // public ICollection<Tour> Tours { get; set; }
        // public ICollection<Attraction> Attractions { get; set; }
    }
}

File: viajes365-frontend/src/app/_models/location.ts
﻿import { Attraction } from './attraction';
import { City } from './city';
import { Photo } from './photo';
import { Tour } from './tour';

export class Location {
  locationId!: number;
  locationName!: string;
  latitude!: number;
  longitude!: number;
  fullAddress!: string;
  note!: string;
  cityId!: number;
  city!: City;
  tours!: Tour[];
  attractions!: Attraction[];
  photos!: Photo[];
  active!: boolean;
  isDeleting = false;
}
